UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/HealthCheck/EventHubLoggingHealthCheck.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+using System.Threading;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.Common.Logging;
+
+namespace UKHO.ExchangeSetService.Common.HealthCheck
+{
+    public class EventHubLoggingHealthCheck : IHealthCheck
+    {
+        private readonly IEventHubLoggingHealthClient eventHubLoggingHealthClient;
+        private readonly ILogger<EventHubLoggingHealthCheck> logger;
+
+        public EventHubLoggingHealthCheck(IEventHubLoggingHealthClient eventHubLoggingHealthClient, ILogger<EventHubLoggingHealthCheck> logger)
+        {
+            this.eventHubLoggingHealthClient = eventHubLoggingHealthClient;
+            this.logger = logger;
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            var healthCheckResult = await eventHubLoggingHealthClient.CheckHealthAsync(context);
+            if (healthCheckResult.Status == HealthStatus.Healthy)
+            {
+                logger.LogInformation(EventIds.EventHubLoggingIsHealthy.ToEventId(), $"Event hub is healthy");
+            }
+            else
+            {
+                logger.LogError(EventIds.EventHubLoggingIsUnhealthy.ToEventId(), $"Event hub is unhealthy responded with error {healthCheckResult.Exception.Message}");
```

Comment:
Please can we use structured logging rather than interpolated string on all log messages.
In this case, please log the exception using the LogError overload that takes the exception as an explicit parameter:
```suggestion
                logger.LogError(EventIds.EventHubLoggingIsUnhealthy.ToEventId(), healthCheckResult.Exception, $"Event hub is unhealthy responded with error {healthCheckResult.Exception.Message}");
```

Reply:
Fixed